src/lib/AccountFinder.ts

```diff
@@ -0,0 +1,58 @@
+import { AccountInfo } from '@/lib/AccountInfo';
+import { NetworkClient } from '@nimiq/network-client';
+import { WalletStore } from '@/lib/WalletStore';
+
+export default class AccountFinder {
+    public static async findAccounts(
+        deriveAccounts: (startIndex: number, count: number) => Promise<Array<{ address: string, keyPath: string }>>,
+        deriveWalletId: (accounts: Array<{ address: string, keyPath: string, balance: number }>) => Promise<string>,
+        defaultLabel: string = 'Standard Account',
+        onIntermediateResult?: (accounts: Array<{ address: string, keyPath: string, balance: number }>) => void,
+    ): Promise<AccountInfo[]> {
+        await NetworkClient.Instance.init(); // initialize NetworkClient if not initialized yet
+        const foundAccounts: Array<{ address: string, keyPath: string, balance: number }> = [];
+        let nextStartIndex = 0;
+        let derivedAccountsPromise = deriveAccounts(nextStartIndex, AccountFinder.MAX_ALLOWED_GAP);
+        nextStartIndex += AccountFinder.MAX_ALLOWED_GAP;
+        while (true) {
+            const derivedAccounts = await derivedAccountsPromise;
+            if (derivedAccounts.length !== AccountFinder.MAX_ALLOWED_GAP) {
+                throw new Error('\'deriveAccounts\' returned wrong number of accounts');
+            }
+
+            // already start deriving next accounts
+            derivedAccountsPromise = deriveAccounts(nextStartIndex, AccountFinder.MAX_ALLOWED_GAP);
+            nextStartIndex += derivedAccounts.length; // by always advancing in blocks of MAX_ALLOWED_GAP it can happen
+            // that we include accounts that are more than MAX_ALLOWED_GAP apart (e.g. one towards the beginning of
+            // the first block, the other one towards the end of the second block). But that's actually not bad.
```

Comment:
I don't understand this comment? Because in the comment you talk about `MAX_ALLOWED_GAP` while the code line actually adds `derivedAccounts.length`.

Reply:
They are bot the same.
This comment is the answer for your remark at line 38.

Reply:
Then at least improve the comment for now, that it's not only `one towards the end` but up to 39 (or `more towards to the end`, and replace `that's not bad` with why it's not bad.